common/include/window.h

```diff
@@ -134,7 +135,14 @@ class window
 	{
 		return wind.prev;
 	}
-	
+
+	bool track(bool *exists)
+	{
+		if (w_exists)
```

Comment:
Since the one caller that uses this does not violate the single-assignment rule, and does not check the return value, it looks like this could be done as an assert instead:
```c++
    void track(bool *e)
    {
        assert(!w_exists);
        w_exists = e;
    }
```
That would make the failure much more obvious in debug builds, and would remove the runtime test in release builds.